backend/analytics_server/mhq/api/teams.py

```diff
@@ -121,9 +121,24 @@ def update_team_repos(team_id: str, repos: List[RawOrgRepo]):
     query_validator = get_query_validator()
     team: Team = query_validator.team_validator(team_id)
 
-    team_repos = get_repository_service().update_team_repos(team, repos)
+    idempotency_key_raw_org_repo_map = {
+        str(repo.idempotency_key): repo for repo in repos
+    }
+
+    team_repos = get_repository_service().update_team_repos(
+        team, repos, idempotency_key_raw_org_repo_map
+    )
```

Comment:
if idempotency_key_raw_org_repo_map is generated from repos data, why are we even passing it down multiple levels of fucns down to the repo layer ? duplicate data being passed down funcs in diff format is not the best solution.

Reply:
makes sense